CrewChiefV4/GameState/GameStateData.cs

```diff
@@ -330,76 +329,91 @@ public void playerStartNewLap(int lapNumber, int position, Boolean inPits, float
         }
 
         public void playerCompleteLapWithProvidedLapTime(int position, float gameTimeAtLapEnd, float providedLapTime,
-            Boolean lapIsValid, Boolean isRaining, float trackTemp, float airTemp, Boolean sessionLengthIsTime, float sessionTimeRemaining)
+            Boolean lapIsValid, Boolean isRaining, float trackTemp, float airTemp, Boolean sessionLengthIsTime, float sessionTimeRemaining, int numberOfSectors)
         {
             if (PlayerLapData.Count > 0)
             {
                 LapData lapData = PlayerLapData[PlayerLapData.Count - 1];
-                playerAddCumulativeSectorData(position, providedLapTime, gameTimeAtLapEnd, lapIsValid, isRaining, trackTemp, airTemp);
-                lapData.LapTime = providedLapTime;
-
-                LapTimePrevious = providedLapTime;
-                if (lapData.IsValid && (PlayerLapTimeSessionBest == -1 || PlayerLapTimeSessionBest > lapData.LapTime))
+                if (PlayerLapData.Count == 1 || !lapData.hasMissingSectors)
```

Comment:
i think you have that logic reversed there. If it's the first lap, *or* it has no missing sectors, then we keep it. Unless I've screw it up?